lighthouse-core/gather/gatherers/dobetterweb/optimized-images.js

```diff
@@ -101,16 +103,31 @@ class OptimizedImages extends Gatherer {
    * @return {!Promise<{originalSize: number, jpegSize: number, webpSize: number}>}
    */
   calculateImageStats(driver, networkRecord) {
-    const param = JSON.stringify(networkRecord.url);
-    const script = `(${getOptimizedNumBytes.toString()})(${param})`;
-    return driver.evaluateAsync(script).then(stats => {
-      const isBase64DataUri = networkRecord.isBase64DataUri;
-      const base64Length = networkRecord.url.length - networkRecord.url.indexOf(',') - 1;
-      return {
-        originalSize: isBase64DataUri ? base64Length : networkRecord.resourceSize,
-        jpegSize: isBase64DataUri ? stats.jpeg.base64 : stats.jpeg.binary,
-        webpSize: isBase64DataUri ? stats.webp.base64 : stats.webp.binary,
-      };
+    let uriPromise = Promise.resolve(networkRecord.url);
+
+    // Get the request body for cross-origin images to circumvent canvas CORS protections
+    if (!networkRecord.isSameOrigin && !networkRecord.isBase64DataUri) {
+      const requestId = networkRecord.requestId;
+      uriPromise = Promise.resolve()
+        .then(_ => driver.sendCommand('Network.enable'))
```

Comment:
> in fact if we disable network properly elsewhere this might not work

the enabled domains are tracked now, so if you enable and then disable, but someone had already disabled before you, it won't actually be disabled.

I've been thinking about putting a check at the end of every pass to yell about gatherers that didn't clean up after themselves by disabling everything they enable :)